package/kedro_viz/server.py

```diff
@@ -213,31 +244,66 @@ def commands():
     type=click.Path(dir_okay=False, writable=True),
     help="Path to save the pipeline JSON file",
 )
-def viz(host, port, browser, load_file, save_file):
+@click.option(
+    "--pipeline",
+    type=str,
+    default=None,
+    help="Name of the modular pipeline to run."
+    "If not set, the default pipeline is visualized",
+)
+@click.option(
+    "--env",
+    "-e",
+    type=str,
+    default=None,
+    multiple=False,
+    envvar="KEDRO_ENV",
+    help="Kedro configuration environment. If not specified, "
+    "catalog config in `local` will be used",
+)
+def viz(host, port, browser, load_file, save_file, pipeline, env):
     """Visualize the pipeline using kedroviz."""
-    _call_viz(host, port, browser, load_file, save_file)
-
-
-def _call_viz(host=None, port=None, browser=None, load_file=None, save_file=None):
+    try:
+        _call_viz(host, port, browser, load_file, save_file, pipeline, env)
+    except KedroCliError:
+        raise
+    except Exception as ex:
+        raise KedroCliError(str(ex))
+
+
+# pylint: disable=too-many-arguments
+def _call_viz(
+    host=None,
+    port=None,
+    browser=None,
+    load_file=None,
+    save_file=None,
+    pipeline_name=None,
+    env=None,
+):
     global data  # pylint: disable=global-statement,invalid-name
 
     if load_file:
-        data = json.loads(Path(load_file).read_text())
-        for key in ["nodes", "edges", "tags"]:
-            if key not in data:
-                click.echo("Invalid file, top level key '{}' not found.".format(key))
-                sys.exit(1)
+        data = _load_from_file(load_file)
     else:
         try:
-            pipeline = get_project_context("create_pipeline")()
-            catalog = get_project_context("create_catalog")(None)
-        except KedroContextError:
-            raise KedroCliError(
-                "Could not find a Kedro project root. "
-                "You can run `kedro viz` by either providing `--load-file` flag with a JSON file "
-                "path for your pipeline, or if the current working directory is "
-                "the root of a Kedro project."
-            )
+            context = get_project_context("context", env=env)
```

Comment:
A thought that just can to me - why do we need this complex logic in the first place if `get_project_context` is a part of the library? You could simply read the current version of Kedro and build the logic based on that, couldn't you?

```python
if kedro.__version__ >= "0.15.2":
    # do this
elif kedro.__version__ >= "0.15.0":
    # do that
else:
    # do something else
```

Reply:
That was the original plan, but installed kedro version in your conda env does not necessary match with the project version. Also `context.project_version` does not necessary match with the installed kedro version. So we decided to to this try/except complex logic.

Reply:
You are running the project with the version of Kedro installed in your current interpreter. `get_project_context` sits in the library, which is indeed updated when you update Kedro. If you updated Kedro and did not update your project, `get_project_context` will break anyways since it makes certain assumptions about the project structure.

Reply:
Aw, I see! I will change it to the above simpler one. That's much better.